sdk/storage/Azure.Storage.Common/swagger/Generator/src/generator.ts

```diff
@@ -479,16 +479,40 @@ function generateOperation(w: IndentWriter, serviceModel: IServiceModel, group:
                     }
                 });
             }
+
+            // We don't throw for 304s on read
+            // (this may be too ambitious for all services, but works for
+            // Storage so I'm not adding x-az- vendor extensions right now)
+            const isReadOperation =
+                operation.method.toLowerCase() == `get` ||
+                operation.method.toLowerCase() == `head`;
+            if (isReadOperation) {
+                w.line(`case 304:`);
+                w.scope('{', '}', () => {
+                    if (result.type === `void`) {
+                        // Just return the response if there's no model
+                        w.line(`return ${responseName};`);
+                    } else {
+                        // Return an exploding Response if there's a model type
+                        // (NoBodyResponse is shared source)
+                        w.line(`return new Azure.NoBodyResponse<${types.getName(result)}>(${responseName});`);
+                    }
+                });
+            }
+
             for (const response of operation.response.failures) {
-                if (response.code === `default`) {
+                if (response.code === `304` && isReadOperation) {
+                    // Ignore 304s handled above
+                    continue;
+                } else if (response.code === `default`) {
                     w.line(`default:`);
                 } else {
                     w.line(`case ${response.code}:`);
                 }
                 w.scope('{', '}', () => {
                     processResponse(response);
-                    if (response.exception) {
-                        // If we're using x-ms-create-exception, we'll pass the response to
+                        if (response.exception) {
```

Comment:
NIT - remove added indent so it's aligned

Reply:
Fixed